File: c++/smallest_rectangle.cpp
// https://programmers.co.kr/learn/courses/30/lessons/86491

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes)
{
    vector<int> larger, smaller;
    for (auto size : sizes)
    {
        if (size[0] < size[1])
        {
            larger.push_back(size[1]);
            smaller.push_back(size[0]);
        }
        else
        {
            smaller.push_back(size[1]);
            larger.push_back(size[0]);
        }
    }
    return (*max_element(larger.begin(), larger.end()) * *max_element(smaller.begin(), smaller.end()));
}

File: python/smallest_rectangle.py
# https://programmers.co.kr/learn/courses/30/lessons/86491?language=python3


def solution(sizes):
    larger = []
    smaller = []
    for size in sizes:
        larger.append(max(size))
        smaller.append(min(size))
    return max(larger) * max(smaller)


# def solution(sizes):
#     return max(max(x) for x in sizes) * max(min(x) for x in sizes)


# def solution(sizes):
#     sizes = [sorted(s) for s in sizes]
#     return max([x[0] for x in sizes]) * max([x[1] for x in sizes])
